src/register/register.component.html

```diff
@@ -1,37 +1,39 @@
-      <a mat-raised-button [routerLink]="['/login']" class="btn btn-link">Login</a>
-    </div>
-  </form>
-</div>
+<h2>Login</h2>
+<form [formGroup]="form" (ngSubmit)="onSubmit(form.value)">
+  <p>
+    <mat-form-field>
+      <input matInput
+             required
+             formControlName="username"
+             placeholder="Username"/>
+      <mat-hint *ngIf="form.controls['username'].hasError('required')">
+        Username is required
+      </mat-hint>
+    </mat-form-field>`
+
+  </p>
+  <p>
+    <mat-form-field>
+      <input matInput
+             required
+             type="password"
+             formControlName="password"
+             placeholder="Password"/>
+      <mat-hint *ngIf="form.controls['password'].hasError('required')">
+        Password is required
+      </mat-hint>
+    </mat-form-field>
+
+  </p>
+  <p>
+    <button type="submit" mat-raised-button color="primary" [disabled]="form.invalid" class="btn btn-primary">
+      Register
+    </button>
+
+    <img *ngIf="loading"
+         alt="Loading"
+         src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA=="/>
+
+    <a mat-raised-button [routerLink]="['/login']" class="btn btn-link">Login</a>
```

Comment:
Register?

Reply:
Login page has a link to register and vise versa